agent/agent_endpoint.go

```diff
@@ -86,7 +99,21 @@ func (s *HTTPServer) AgentMetrics(resp http.ResponseWriter, req *http.Request) (
 	if rule != nil && !rule.AgentRead(s.agent.config.NodeName) {
 		return nil, acl.ErrPermissionDenied
 	}
+	if enablePrometheusOutput(req) {
+		if s.agent.config.TelemetryPrometheusRetentionTime.Nanoseconds() < 1 {
```

Comment:
nitpick: time.Duration is really just an int anyway so you can just do `s.agent.config.TelemetryPrometheusRetentionTime < 1` here.